src/app/zap-templates/zcl/data-model/chip/energy-evse-cluster.xml

```diff
@@ -0,0 +1,235 @@
+    <item name="ChargingEnabled"       value="0x01"  optional="false"/>
+    <item name="DischargingEnabled"    value="0x02"  optional="true" />
+    <item name="DisabledError"         value="0x03"  optional="false"/>
+    <item name="DisabledDiagnostics"   value="0x04"  optional="false"/>
+  </enum>
+
+  <enum name="FaultStateEnum" type="ENUM8">
+    <cluster code="0x0099"/>
+    <item name="NoError"            value="0x00"  optional="false"/>
+    <item name="EvseMeterFailure"   value="0x01"  optional="false"/>
+    <item name="OverVoltage"        value="0x02"  optional="false" />
+    <item name="UnderVoltage"       value="0x03"  optional="false"/>
+    <item name="OverCurrent"        value="0x04"  optional="false"/>
+    <item name="ContactWetFailure"  value="0x05"  optional="false"/>
+    <item name="ContactDryFailure"  value="0x06"  optional="false"/>
+    <item name="GroundFault"        value="0x07"  optional="false"/>
+    <item name="PowerLoss"          value="0x08"  optional="false"/>
+    <item name="PowerQuality"       value="0x09"  optional="false"/>
+    <item name="PilotShortCircuit"  value="0x0A"  optional="false"/>
+    <item name="EmergencyStop"      value="0x0B"  optional="false"/>
+    <item name="EvDisconnected"     value="0x0C"  optional="false"/>
+    <item name="WrongPowerSupply"   value="0x0D"  optional="false"/>
+    <item name="LiveNeutralSwap"    value="0x0E"  optional="false"/>
+    <item name="OverTemperature"    value="0x0F"  optional="false"/>
+    <item name="Other"              value="0xFF"  optional="false"/>
+  </enum>
+
+  <bitmap name="TargetDayOfWeekBitmap" type="BITMAP8">
+    <cluster code="0x0099"/>
+    <field name="Sunday" mask="0x01"/>
+    <field name="Monday" mask="0x02"/>
+    <field name="Tuesday" mask="0x04"/>
+    <field name="Wednesday" mask="0x08"/>
+    <field name="Thursday" mask="0x10"/>
+    <field name="Friday" mask="0x20"/>
+    <field name="Saturday" mask="0x40"/>
+    <field name="Away" mask="0x80"/>
+  </bitmap>
+
+  <enum name="StartOfWeekEnum" type="ENUM8">
+    <cluster code="0x0099"/>
+    <item name="Sunday" value="0x00"/>
+    <item name="Monday" value="0x01"/>
+    <item name="Tuesday" value="0x02"/>
+    <item name="Wednesday" value="0x03"/>
+    <item name="Thursday" value="0x04"/>
+    <item name="Friday" value="0x05"/>
+    <item name="Saturday" value="0x06"/>
+  </enum>
+
+  <enum name="EnergyTransferStoppedReasonEnum" type="ENUM8">
+    <cluster code="0x0099"/>
+    <item name="EvStopped"    value="0x00"  optional="false"/>
+    <item name="EvseStopped"  value="0x01"  optional="false"/>
+    <item name="Other"        value="0x03"  optional="false"/>
+  </enum>
+
+  <struct name="ChargingTargetStruct">
+    <cluster code="0x0099"/>
+    <item name="TargetTime"   type="INT16U"  optional="false"/>
+    <item name="TargetSoC"    type="INT8U"   optional="true"/>
+    <item name="AddedEnergy"  type="INT32U"  optional="true"/>
```

Comment:
the type here seems to be `Energy` in the spec. I am unclear where it says that energi is a u32 ... could we update the spec for that? Maybe we should not use ref_Energy and instead put this as a constraint instead.

Reply:
Hmm thought I had added one. But upon reflection I think this belongs in a common data types area - so have added a question on slack to consider if we put one there or add it to this.
If it is common then it probably needs to be an sint32 not uint32 and we'd call it `energy-wh`?